app/assets/javascripts/modules/feedback.js

```diff
@@ -9,41 +9,76 @@
+      }
+
+      // Form containers
+      const formContainers = feedback.querySelectorAll('[aria-controls]')
+
+      // Form close button
+      const closeButtonTemplate = document.getElementById('feedback-close-button-template')
+      const closeButtonClone = closeButtonTemplate.content.cloneNode(true)
+
+      // Form toggling
+      let formId
+      let formContainer
+      let formCloseToggle
+      let formToggle
       let i
-      let target
-      for (i = 0; i < toggles.length; i = i + 1) {
-        toggleID = toggles[i].getAttribute('aria-controls')
-        toggleContent = document.getElementById(toggleID)
-        toggleContent.setAttribute('tabindex', '-1')
+      for (i = 0; i < formContainers.length; i = i + 1) {
+        formId = formContainers[i].getAttribute('aria-controls')
+        formContainer = document.getElementById(formId)
+        formContainer.setAttribute('tabindex', '-1')
+
+        // Add close button to each form before fieldset (and after any error summary)
+        const formFieldset = formContainer.querySelector('.govuk-fieldset')
+        formContainer.insertBefore(closeButtonClone, formFieldset)
+        formCloseToggle = formContainer.querySelector('.app-feedback__button--close')
       }
 
-      function toggle (ev) {
+      function toggleForm (ev) {
         ev = ev || window.event
-        target = ev.target || ev.srcElement
-        if (target.hasAttribute('aria-controls')) {
-          toggleID = target.getAttribute('aria-controls')
-          toggleContent = document.getElementById(toggleID)
-
-          if (toggleContent.getAttribute('aria-hidden') === 'true') {
-            toggleContent.setAttribute('aria-hidden', 'false')
-            target.setAttribute('aria-expanded', 'true')
-            toggleQuestion.setAttribute('aria-hidden', 'true')
-            toggleQuestion.setAttribute('aria-expanded', 'false')
+        formToggle = ev.target || ev.srcElement
+        if (formToggle.hasAttribute('aria-controls')) {
+          formId = formToggle.getAttribute('aria-controls')
+          formContainer = document.getElementById(formId)
+
+          if (formContainer.getAttribute('aria-hidden') === 'true') {
+            formToggle.setAttribute('aria-expanded', 'true')
+            formCloseToggle.setAttribute('aria-expanded', 'true')
+            formContainer.setAttribute('aria-hidden', 'false')
+            promptContainer.classList.add('govuk-visually-hidden')
           } else {
-            toggleContent.setAttribute('aria-hidden', 'true')
-            target.setAttribute('aria-expanded', 'false')
-            toggleQuestion.setAttribute('aria-hidden', 'false')
-            toggleQuestion.setAttribute('aria-expanded', 'true')
+            formContainer.setAttribute('aria-hidden', 'true')
+            formCloseToggle.setAttribute('aria-expanded', 'false')
+            formToggle.setAttribute('aria-expanded', 'false')
+            promptContainer.classList.remove('govuk-visually-hidden')
```

Comment:
Are the polyfills available for this here? (I've asked on x-gov)

Reply:
Have added a test for `window.addEventListener`, `window.DOMTokenList` (for `classList`) and `window.HTMLTemplateElement` (for `<template>`s). If any of these tests fail, the simpler feedback form will be shown. I think that’s a good compromise given the widespread support for these methods.